File: src/php/parsers/FreeBestTv.php
<?php

namespace zikwall\m3uparse\parsers;

use zikwall\m3uparse\base\BaseParse;
use zikwall\m3uparse\Helper;
use zikwall\m3uparse\interfaces\IParse;
use zikwall\m3uparse\Playlists;

class FreeBestTv extends BaseParse implements IParse
{
    public function parse()
    {
        $sourceUrl = 'http://4pda.ru/pages/go/?u=http%3A%2F%2Ftopplay.do.am%2FFreeBestTV.m3u&e=84875135';
        Helper::download(Helper::getPlaylistUploadDir(), 'free_best_tv', $sourceUrl);

        $source = file_get_contents(Playlists::get('free_best_tv'));
        $items = explode("#EXTINF:-1,", $source);
        $items = array_slice($items, 3);

        $playlist = [];

        foreach ($items as $item) {
            if (strpos($item, 'https') === false) {
                continue;
            }

            list($name, $url) = preg_split('/\r\n|\r|\n/', $item);

            $playlist[] = [
                'name' => trim($name),
                'url'  => trim($url),
                'from' => 'free_best_tv'
            ];
        }

        return $playlist;
    }

    public function channels() : array
    {
        return json_decode(file_get_contents(dirname(__DIR__) . '/default/free.json'), true);
    }
}

File: src/go/parsers/free_best_tv.go
package parsers

import (
	parse "../../go"
	"io/ioutil"
	"strings"
)

func FreeBEstTv() []parse.Playlist {
	if err := parse.DownloadFile("./plists/uploads/free_best_tv.m3u", parse.URLS["free_best_tv"]); err != nil {
		panic(err)
	}

	b, err := ioutil.ReadFile("./plists/uploads/free_best_tv.m3u")

	if err != nil {
		panic(err)
	}

	s := strings.Split(string(b), "#EXTINF:-1,")
	result := []parse.Playlist{}

	for _, pl := range s[3:] {
		item := strings.Split(pl, "\n")
		name := strings.TrimSpace(item[0])
		url := strings.TrimSpace(item[1])

		if !strings.Contains(url, "http") {
			continue
		}

		result = append(result, parse.Playlist{
			Name: name,
			Url:  url,
			From: "free_best_tv",
		})
	}

	return result
}
